keycloak-iam/src/main/java/com/valtimo/keycloak/service/KeycloakUserManagementService.java

```diff
@@ -266,13 +276,13 @@ private List<RoleRepresentation> getRolesFromUser(UserRepresentation userReprese
             var realmRoles = keycloakService
                 .usersResource(keycloak)
                 .get(userRepresentation.getId())
-                .roles().realmLevel().listAll();
+                .roles().realmLevel().listEffective(true);
             var roles = new ArrayList<>(realmRoles);
             if (!clientName.isBlank()) {
                 var clientRoles = keycloakService
                     .usersResource(keycloak)
                     .get(userRepresentation.getId())
-                    .roles().clientLevel(keycloakService.getClientId(keycloak)).listAll();
+                    .roles().clientLevel(keycloakService.getClientId(keycloak)).listEffective(true);
```

Comment:
@theo-ritense I've made some changes.

Here the `listEffective(true)` will make sure to also retrieve group roles. I've also enabled `briefRepresentation` where possible.